Assets/Src/Scripts/SeedCalc/Calculator.cs

```diff
@@ -143,105 +232,229 @@ private class Visualizer : IVisualizer<BinaryEvent>, IVisualizer<EvalEvent> {
           case CalculatorInput.Equal:
             Execute();
             break;
-          case CalculatorInput.Dot:
-            // Adds the leading zero before a single dot.
-            if (_cachedInput.Length <= 0) {
-              Append(CalculatorInput.N0);
-            }
-            Append(CalculatorInput.Dot);
-            if (State.IsOk()) {
-              Parse();
-            }
-            break;
           default:
-            Append(input);
-            if (State.IsOk()) {
-              Parse();
-            }
+            HandleVisibleKeys(input);
             break;
         }
       }
     }
 
+    private void HandleVisibleKeys(string input) {
+      switch (input) {
+        case CalculatorInput.Dot:
+          // Do not repeat dots.
+          char lastChar = !(ParsedExpression is null) && ParsedExpression.Expression.Length > 0 ?
+              ParsedExpression.Expression[ParsedExpression.Expression.Length - 1] : '\0';
+          if (lastChar != CalculatorInput.Dot[0]) {
+            // Adds a leading zero if "." is the first input of a number.
+            if (ParsedExpression is null || !ParsedExpression.LastTokenIsNumber()) {
+              Append(CalculatorInput.N0);
+            }
+            Append(input);
+          }
+          break;
+        case CalculatorInput.N0:
+          // A number can be "0", but cannot be "0123".
+          if (ParsedExpression is null || !ParsedExpression.LastTokenIsNumber()) {
+            Append(input);
+          } else if (ParsedExpression.ExtractLastTokenText() != "0") {
+            Append(input);
+          }
+          break;
+        case CalculatorInput.N00:
+          // A number can be neither "00" nor "00123".
+          if (ParsedExpression is null || !ParsedExpression.LastTokenIsNumber()) {
+            // To start a number, only one "0" is appended.
+            Append(CalculatorInput.N0);
+          } else if (ParsedExpression.ExtractLastTokenText() != "0") {
+            Append(input);
+          }
+          break;
+        case CalculatorInput.N1:
+        case CalculatorInput.N2:
+        case CalculatorInput.N3:
+        case CalculatorInput.N4:
+        case CalculatorInput.N5:
+        case CalculatorInput.N6:
+        case CalculatorInput.N7:
+        case CalculatorInput.N8:
+        case CalculatorInput.N9:
+          if (!(ParsedExpression is null) && ParsedExpression.ExtractLastTokenText() == "0") {
+            // If there is only one "0" in the current number, replaces it with the next non-zero
```

Comment:
The comment is not correct? Shall be backspace when the last text is "0"?

Reply:
Done. The comments are updated to a clearer sentence.